spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,579 @@
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.builder.Roles;
+import org.candlepin.spec.bootstrap.data.builder.Users;
+import org.candlepin.spec.bootstrap.data.util.ExportUtil;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.collections.CollectionUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+@SpecTest
+class ExportSpecTest {
+    private static final String EXPORT_JOB_KEY = "ExportJob";
+    private static final String RECORD_CLEANER_JOB_KEY = "ImportRecordCleanerJob";
+    private static final String UNDO_IMPORTS_JOB_KEY = "UndoImportsJob";
+    /** The following are directory location paths within a manifest */
+    private static final String EXPORT_PATH = "export";
+    private static final String PRODUCTS_PATH = "export/products";
+    private static final String ENTITILEMENTS_PATH = "export/entitlements";
+    private static final String ENTITILEMENT_CERTIFICATES_PATH =  "export/entitlement_certificates";
+    private static final String DISTRIBUTOR_VERSION_PATH = "export/distributor_version";
+    private static final String CONSUMER_TYPE_PATH = "export/consumer_types";
+    private static final String CDN_PATH = "export/content_delivery_network";
+    private static final String RULES_PATH = "export/rules2";
+
+    @Test
+    @DisplayName("should allow manifest creation with read only user")
+    void shouldAllowManifestCreationWithReadOnlyUser() throws Exception {
+        ApiClient client = ApiClients.admin();
+        OwnerDTO owner = client.owners().createOwner(Owners.random());
+        ConsumerDTO readOnlyConsumer = createReadOnlyConsumer(client, owner);
+
+        File readOnlyManifest = createExport(client, readOnlyConsumer.getUuid(), null);
+        ZipFile readOnlyExport = ExportUtil.getExportArchive(readOnlyManifest);
+        List<ZipEntry> exportDir = ExportUtil.getFilesInDirectory(EXPORT_PATH, readOnlyExport);
+        List<ZipEntry> consumerFiles = exportDir.stream()
+            .filter(e -> e.getName().contains("consumer.json"))
```

Comment:
That makes sense! I updated this throughout to not use 'e' as the single character name and replaced the name comparison to use `endsWith()` instead of `contains`. I applied this to other tests as well!

For the verification that the entry is a file, I found out that all the entries in the ZipFile are files, it doesn't have entries for just directories, so all the entries `isDirectory()` will return false.